### PR TITLE
Remove milestone related types and functions

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: bindings/core/src/response.rs [hunk 1]
```diff
@@ -144,10 +141,6 @@ pub enum Response {
     /// - [`FindOutputs`](crate::method::ClientMethod::FindOutputs)
     Outputs(Vec<OutputWithMetadataResponse>),
     /// Response for:
-    /// - [`GetMilestoneById`](crate::method::ClientMethod::GetMilestoneById)
-    /// - [`GetMilestoneByIndex`](crate::method::ClientMethod::GetMilestoneByIndex)
-    Milestone(MilestonePayloadDto),
-    /// Response for:
     /// - [`GetMilestoneByIdRaw`](crate::method::ClientMethod::GetMilestoneByIdRaw)
     /// - [`GetMilestoneByIndexRaw`](crate::method::ClientMethod::GetMilestoneByIndexRaw)
     MilestoneRaw(Vec<u8>),
```
Comment (Thoralf-M) [inline bindings/core/src/response.rs]: Also remove, I know it's a draft :trollface:

Comment (thibault-martinez) [reply bindings/core/src/response.rs]: Whoops

Comment (thibault-martinez) [reply bindings/core/src/response.rs]: Done

File: sdk/examples/client/block/01_block_confirmation_time.rs [hunk 2]
```diff
@@ -34,17 +34,17 @@ async fn main() -> Result<()> {
...
-        let ms = client.get_milestone_by_index(ms_index).await?;
-        println!(
-            "Block {block_id} got confirmed by milestone {ms_index} at timestamp {}.",
-            ms.essence().timestamp()
-        );
-    } else {
-        println!("Block {block_id} is not confirmed.")
-    }
+    // let metadata = client.get_block_metadata(&block_id).await?;
+
```
Comment (kwek20) [inline sdk/examples/client/block/01_block_confirmation_time.rs:39]: you left this  commented

File: sdk/tests/client/node_api.rs [hunk 3]
```diff
@@ -343,27 +265,18 @@ async fn test_mqtt() {
...
-                    Ok(_) | Err(TrySendError::Full(_)) => (),
-                    e => e.unwrap(),
+        .subscribe([Topic::new("blocks").unwrap()], move |evt| {
+            match &evt.payload {
+                MqttPayload::Block(_) => {
+                    assert_eq!(evt.topic, "blocks");
                 }
-            },
-        )
+                MqttPayload::Json(_) => {}
```
Comment (kwek20) [inline sdk/tests/client/node_api.rs]: we should panic here like before: `_ => panic!("unexpected mqtt payload type: {:?}", evt),` as we are defenitely not eexpecting json whhen only requesting block right?

File: sdk/tests/types/transaction_regular_essence.rs [hunk 4]
```diff
@@ -108,30 +105,30 @@ fn build_valid_add_inputs_outputs() {
...
-        .with_inputs([input1, input2])
-        .add_output(output)
-        .with_payload(rand_milestone_payload(42))
-        .finish(&protocol_parameters);
-
-    assert!(matches!(essence, Err(Error::InvalidPayloadKind(7))));
-}
+// #[test]
+// fn build_invalid_payload_kind() {
+//     let protocol_parameters = protocol_parameters();
```
Comment (kwek20) [inline sdk/tests/types/transaction_regular_essence.rs]: same here, leave a todo maybe?

File: sdk/examples/client/block/01_block_confirmation_time.rs (hunk 2 above)
Comment (thibault-martinez) [reply sdk/examples/client/block/01_block_confirmation_time.rs:39]: Done

File: sdk/tests/client/node_api.rs (hunk 3 above)
Comment (thibault-martinez) [reply sdk/tests/client/node_api.rs]: Done

File: sdk/tests/types/transaction_regular_essence.rs (hunk 4 above)
Comment (thibault-martinez) [reply sdk/tests/types/transaction_regular_essence.rs]: Done

File: sdk/src/types/api/core/response.rs [hunk 5]
```diff
@@ -249,34 +248,6 @@ pub enum OutputResponse {
...
-
-/// Response of GET /api/core/v2/milestone/{milestone_index}/utxo-changes.
-/// Returns all UTXO changes that happened at a specific milestone.
-#[derive(Clone, Debug, Eq, PartialEq)]
-#[cfg_attr(
-    feature = "serde",
-    derive(serde::Serialize, serde::Deserialize),
-    serde(rename_all = "camelCase")
-)]
-pub struct UtxoChangesResponse {
```
Comment (Thoralf-M) [inline sdk/src/types/api/core/response.rs:274]: Will this not exist for confirmed slots?

File: sdk/src/types/block/error.rs [hunk 6]
```diff
@@ -254,21 +235,6 @@ impl fmt::Display for Error {
             Self::MigratedFundsNotSorted => {
```
Comment (Thoralf-M) [inline sdk/src/types/block/error.rs]: Should also be removed

File: sdk/src/types/block/error.rs [hunk 7]
```diff
@@ -197,11 +183,6 @@ impl fmt::Display for Error {
             Self::InvalidMetadataFeatureLength(length) => {
                 write!(f, "invalid metadata feature length {length}")
             }
-            Self::InvalidMilestoneMetadataLength(length) => {
-                write!(f, "invalid milestone metadata length {length}")
-            }
-            Self::InvalidMilestoneOptionCount(count) => write!(f, "invalid milestone option count: {count}"),
-            Self::InvalidMilestoneOptionKind(k) => write!(f, "invalid milestone option kind: {k}"),
             Self::InvalidMigratedFundsEntryAmount(amount) => {
```
Comment (Thoralf-M) [inline sdk/src/types/block/error.rs]: Also remove?

File: sdk/src/types/api/core/response.rs (hunk 5 above)
Comment (thibault-martinez) [reply sdk/src/types/api/core/response.rs:274]: Probably but I'll just reintroduce it the right way, there was a lot of reliance on Milestone stuff so I'm just removing it for now, easier

File: sdk/src/types/block/error.rs (hunk 6 above)
Comment (thibault-martinez) [reply sdk/src/types/block/error.rs]: Done

File: sdk/src/types/block/error.rs (hunk 7 above)
Comment (thibault-martinez) [reply sdk/src/types/block/error.rs]: Done

File: sdk/src/types/block/error.rs [hunk 8]
```diff
@@ -90,11 +82,6 @@ pub enum Error {
...
     InvalidUnlockConditionKind(u8),
     InvalidFoundryZeroSerialNumber,
-    MigratedFundsNotSorted,
-    MilestoneInvalidSignatureCount(<SignatureCount as TryFrom<usize>>::Error),
-    MilestonePublicKeysSignaturesCountMismatch { key_count: usize, sig_count: usize },
-    MilestoneOptionsNotUniqueSorted,
-    MilestoneSignaturesNotUniqueSorted,
     MissingAddressUnlockCondition,
     MissingGovernorUnlockCondition,
     MissingPayload,
```
Comment (Thoralf-M) [inline sdk/src/types/block/error.rs:87]: Can't comment there, but `TailTransactionHashNotUnique` is also only required for milestones

Comment (thibault-martinez) [reply sdk/src/types/block/error.rs:87]: Done

**Review by Thoralf-M — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by qrayven — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**